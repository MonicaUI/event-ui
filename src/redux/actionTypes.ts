export const LOAD_ACCOUNT_SNAPSHOT = 'event-guestside-site/LOAD_ACCOUNT_SNAPSHOT';
export const LOAD_EVENT_SNAPSHOT = 'event-guestside-site/LOAD_EVENT_SNAPSHOT';
export const LOAD_WEBSITE_CONTENT = 'event-guestside-site/LOAD_WEBSITE_CONTENT';
export const SET_REFERRER = 'event-guestside-site/SET_REFERRER';
export const LOAD_REGISTRATION_PROCESS_CONTENT = 'event-guestside-site/LOAD_REGISTRATION_PROCESS_SNAPSHOT';
export const LOAD_VISIBLE_PRODUCTS = 'event-guestside-site/LOAD_VISIBLE_PRODUCTS';
export const LOAD_REG_CART_VISIBLE_PRODUCTS = 'event-guestside-site/LOAD_REG_CART_VISIBLE_PRODUCTS';
export const REPLACE_VISIBLE_SESSION_PRODUCTS = 'event-guestside-site/REPLACE_VISIBLE_SESSION_PRODUCTS';
export const LOAD_VISIBLE_SESSION_PRODUCTS = 'event-guestside-site/LOAD_VISIBLE_SESSION_PRODUCTS';
export const LOAD_VISIBLE_PRODUCTS_BY_REG_TYPE = 'event-guestside-site/LOAD_VISIBLE_PRODUCTS_BY_REG_TYPE';
export const LOAD_EVENT_WEB_APP_SETTINGS = 'event-guestside-site/LOAD_EVENT_WEB_APP_SETTINGS';
export const SET_WEBSITE_PASSWORD_VALIDITY = 'event-guestside-site/SET_WEBSITE_PASSWORD_VALIDITY';
export const UPDATE_QUESTION_VISIBILTY = 'event-guestside-site/UPDATE_QUESTION_VISIBILITY';
export const GET_ORDERS = 'event-guestside-site/GET_ORDERS';
export const CLEAR_ORDERS = 'event-guestside-site/CLEAR_ORDERS';
export const SWITCH_CCPA_DIALOG_MODE = 'event-guestside-site/SWITCH_CCPA_DIALOG_MODE';
export const SHOW_TRANSPARENT_WRAPPER = 'event-guestside-site/SHOW_TRANSPARENT_WRAPPER';
export const HIDE_TRANSPARENT_WRAPPER = 'event-guestside-site/HIDE_TRANSPARENT_WRAPPER';
export const SPINNER_SELECTION_PENDING = 'event-guestside-site/SPINNER_SELECTION_PENDING';
export const SPINNER_SELECTION_DONE = 'event-guestside-site/SPINNER_SELECTION_DONE';
export const RESTORE_ALL_REG_TYPES_FOR_EVENT = 'event-guestside-site/RESTORE_ALL_REG_TYPES_FOR_EVENT';
