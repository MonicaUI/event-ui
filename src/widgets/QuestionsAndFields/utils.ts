import { getIn } from 'icepick';
import getJSONValue from 'nucleus-widgets/utils/fields/getJSONValue';
import { getContactFieldForWidget } from 'event-widgets/redux/selectors/website/pluginData/registrationProcessNavigation';
import { getSubQuestionParentQuestionIdJsonPath } from 'event-widgets/redux/selectors/appData';
import { NAChoice, OtherChoice } from '@cvent/nucleus-form-validations';

const DEFAULT_QUESTION_CHOICES = Object.freeze([]);

export function getConfigWithDisplay(config: $TSFixMe, display: $TSFixMe): $TSFixMe {
  return {
    ...config,
    appData: {
      ...config.appData,
      display
    }
  };
}

export function convertChoicesToBaseLanguage(
  choices: $TSFixMe,
  choicesInCurrentLanguage: $TSFixMe,
  choicesInBaseLanguage: $TSFixMe
): $TSFixMe {
  return choices
    .map(choice => {
      const localizedChoice = choicesInCurrentLanguage.find(c => c.text === choice);
      if (localizedChoice) {
        const baseLangChoice = choicesInBaseLanguage.find(c => c.id === localizedChoice.id);
        if (baseLangChoice) {
          return {
            choice: baseLangChoice.text,
            answerType: 'Choice'
          };
        }
      } else {
        return {
          choice,
          answerType: 'Choice'
        };
      }
    })
    .filter(Boolean);
}

const emptyOrNaOrOther = value => !value || value === NAChoice || value === OtherChoice;

const needsTranslation = (baseChoices, localizedChoices, text) =>
  !localizedChoices.find(c => c.text === text) && baseChoices.find(c => c.text === text);

export function setSelectedValuesForChoices(
  answer: $TSFixMe,
  baseChoices: $TSFixMe,
  localizedChoices: $TSFixMe
): $TSFixMe {
  const translateEachIfNecessary = selectedValues =>
    selectedValues
      .filter(value => !emptyOrNaOrOther(value))
      .map(text =>
        needsTranslation(baseChoices, localizedChoices, text)
          ? updateSelectedValues(text, baseChoices, localizedChoices)
          : text
      );
  const selectedValues = translateEachIfNecessary(answer.value.selectedValues || []);
  const selectedPrevValues = translateEachIfNecessary(getIn(answer, ['valueBeforeMod', 'selectedValues']) || []);

  addMissingChoicesToLocaledAnswer(selectedValues, selectedPrevValues, answer);

  return {
    ...answer,
    value: {
      ...answer.value,
      selectedValues: selectedValues.length ? selectedValues : answer.value.selectedValues || []
    },
    valueBeforeMod: {
      ...answer.valueBeforeMod,
      selectedValues: selectedPrevValues.length
        ? selectedPrevValues
        : getIn(answer, ['valueBeforeMod', 'selectedValues']) || []
    }
  };
}

function addOtherOrNAChoiceToLocalizedAnswer(selectedValues, answerSelectedValues, otherOrNAChoice) {
  if (selectedValues.length && answerSelectedValues.includes(otherOrNAChoice)) {
    selectedValues.push(otherOrNAChoice);
  }
}

function addMissingChoicesToLocaledAnswer(selectedValues, selectedPrevValues, answer) {
  const answerSelectedValues = getIn(answer, ['value', 'selectedValues']) || [];
  const answerSelectedPrevValues = getIn(answer, ['valueBeforeMod', 'selectedValues']) || [];
  addOtherOrNAChoiceToLocalizedAnswer(selectedValues, answerSelectedValues, NAChoice);
  addOtherOrNAChoiceToLocalizedAnswer(selectedPrevValues, answerSelectedPrevValues, NAChoice);
  addOtherOrNAChoiceToLocalizedAnswer(selectedValues, answerSelectedValues, OtherChoice);
  addOtherOrNAChoiceToLocalizedAnswer(selectedPrevValues, answerSelectedPrevValues, OtherChoice);
}

function updateSelectedValues(choiceText, baseChoices, localizedChoices) {
  const choice = baseChoices.find(c => c.text === choiceText);
  if (choice) {
    const updatedChoice = localizedChoices.find(c => c.id === choice.id);
    if (updatedChoice) {
      return updatedChoice.text;
    }
  }
}

export function getLocalizedContactFieldForWidget(state: $TSFixMe, config: $TSFixMe, id: $TSFixMe): $TSFixMe {
  const registrationField = getContactFieldForWidget(state, config.registrationFieldPageType, config.fieldId, id);
  return {
    ...registrationField,
    displayName: getIn(config, ['appData', 'displayName']) || registrationField.displayName
  };
}

export function modifyFieldDefinitionForMultiLanguage(fieldDefinition: $TSFixMe): $TSFixMe {
  const modifiedFieldDefinition = {
    ...fieldDefinition,
    question: {
      ...fieldDefinition.question,
      questionTypeInfo: {
        ...fieldDefinition.question.questionTypeInfo,
        choices: fieldDefinition.question.questionTypeInfo.choices.map(choice => {
          return { id: choice.text.slice(choice.text.indexOf('.') + 1), text: choice.text };
        })
      }
    }
  };
  return modifiedFieldDefinition;
}

export function setParentQuestionId(appData: $TSFixMe, config: $TSFixMe, questionId: $TSFixMe): $TSFixMe {
  return {
    ...config,
    appData: {
      ...config.appData,
      parentQuestionId: getJSONValue(appData, getSubQuestionParentQuestionIdJsonPath(appData, questionId))
    }
  };
}

export function getQuestionCategory(question: $TSFixMe): $TSFixMe {
  if (question.isProductQuestion) {
    return 'productQuestions';
  }
  if (question.isTravelQuestion) {
    return 'travelQuestions';
  }
  return 'registrationQuestions';
}

export function setDefaultChoicesPreSelected(
  appData: $TSFixMe,
  questionCategory: $TSFixMe,
  questionId: $TSFixMe,
  answer: $TSFixMe,
  baseChoices: $TSFixMe
): $TSFixMe {
  const questionDefaultChoices = getIn(
    appData,
    ['registrationSettings', questionCategory, questionId, 'question', 'questionTypeInfo', 'defaultChoice'],
    DEFAULT_QUESTION_CHOICES
  );
  let filteredChoices = DEFAULT_QUESTION_CHOICES;
  if (questionDefaultChoices && questionDefaultChoices.length > 0) {
    if (!Array.isArray(questionDefaultChoices)) {
      filteredChoices = baseChoices.filter(c => c.id === questionDefaultChoices).map(choice => choice.text);
    } else {
      filteredChoices = baseChoices.map(choice => {
        const matchIndex = questionDefaultChoices.findIndex(defaultChoice => defaultChoice.id === choice.id);
        if (matchIndex !== -1) {
          return choice.text;
        }
      });
    }
    filteredChoices = filteredChoices.filter(x => x !== undefined);
    return {
      ...answer,
      value: {
        ...answer.value,
        selectedValues: filteredChoices
      },
      valueBeforeMod: {
        ...answer.valueBeforeMod,
        selectedValues: filteredChoices
      }
    };
  }
  return {
    ...answer,
    value: {
      ...answer.value,
      useDefaultChoices: false // answer = undefined and Question does not have defautl choices
    }
  };
}
