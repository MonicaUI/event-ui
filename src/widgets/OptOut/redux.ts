/* eslint-disable import/no-cycle */

import { injectPageSnapshot } from '../../redux/actions';
import querystring from 'querystring';

/**
 * Opt-out and opt-in Redux module:
 * Event invitees can click "Opt Out" link in their emails to opt them out all communications from the event account
 */
const GET_OPT_OUT_STATUS = 'event-guestside-site/optOut/GET_OPT_OUT_STATUS';
const OPT_OUT_CONTACT = 'event-guestside-site/optOut/OPT_OUT_CONTACT';

const createGetOptedOutStatusAction = contactInfo => {
  return {
    type: GET_OPT_OUT_STATUS,
    payload: contactInfo
  };
};

export const getOptOutStatus =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventGuestClient },
      accessToken,
      event: { id: eventId }
    } = getState();
    let {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    if (contactId) {
      inviteeId = null;
    }
    const optedOutContact = await eventGuestClient.getOptOutStatus(accessToken, eventId, inviteeId, guestId, contactId);
    dispatch(createGetOptedOutStatusAction(optedOutContact));
  };

export const createOptedOutAction = (contactInfo: $TSFixMe): $TSFixMe => {
  return {
    type: OPT_OUT_CONTACT,
    payload: contactInfo
  };
};

export const optOutAction =
  (optOutStatus: $TSFixMe, fromConfirmPage: $TSFixMe): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventGuestClient },
      accessToken,
      event: { id: eventId }
    } = getState();
    let {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    if (contactId) {
      inviteeId = null;
    }
    // eslint-disable-next-line import/no-cycle
    const page = (await import(/* webpackChunkName optOut */ './content')).optInOutPageNew;
    if ((fromConfirmPage && optOutStatus) || !fromConfirmPage) {
      const optedOutContact = await eventGuestClient.optOut(
        accessToken,
        eventId,
        inviteeId,
        guestId,
        contactId,
        optOutStatus
      );
      dispatch(createOptedOutAction(optedOutContact));
    }
    dispatch(injectPageSnapshot(page));
  };

export const prepareForOptOutPageLoad =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    let page;
    if (!inviteeId && !guestId && !contactId) {
      page = (await import(/* webpackChunkName optOut */ './invalidInviteeContent')).invalidInviteeOptOut;
    } else {
      await dispatch(getOptOutStatus());
      if (getState() && getState().optOut && getState().optOut.optOutStatus) {
        // eslint-disable-next-line import/no-cycle
        page = (await import(/* webpackChunkName optOut */ './content')).optInOutPage;
      } else {
        // eslint-disable-next-line import/no-cycle
        page = (await import(/* webpackChunkName optOut */ './content')).confirmationPage;
      }
    }
    dispatch(injectPageSnapshot(page));
  };

export const optOutReducer = (state = {}, action: $TSFixMe): $TSFixMe => {
  switch (action.type) {
    case GET_OPT_OUT_STATUS:
    case OPT_OUT_CONTACT:
      return {
        ...state,
        ...action.payload
      };
    default:
      return state;
  }
};
