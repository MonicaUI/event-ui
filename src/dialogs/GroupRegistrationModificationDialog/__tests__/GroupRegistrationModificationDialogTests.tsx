import React from 'react';
import createStoreWithMiddleware from 'nucleus-guestside-site/src/redux/devCreateStoreWithMiddleware';
import { openGroupRegistrationModificationDialog } from '..';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import DialogContainer from 'nucleus-guestside-site/src/containers/DialogContainer';
import { combineReducers } from 'redux';
import dialogContainer from 'nucleus-guestside-site/src/redux/modules/dialogContainer';
import EventSnapshot from '../../../../fixtures/EventSnapshot.json';
import { InviteeStatusId } from 'event-widgets/utils/InviteeStatus';

const state = {
  website: EventSnapshot.eventSnapshot.siteEditor.website,
  text: {
    translate: (resx, tokens) => (tokens ? `${resx}:${JSON.stringify(tokens)}` : resx)
  },
  appData: {
    registrationSettings: {
      registrationPaths: {
        regPathId1: {
          modification: {
            enabled: true
          }
        }
      }
    }
  },
  registrationForm: {
    regCart: {
      eventRegistrations: {
        '0bc2a08e-cde2-45f6-bc7f-d0f677013668': {
          attendee: {
            personalInformation: {
              firstName: 'Ian',
              lastName: 'Garcia'
            },
            inviteeStatus: InviteeStatusId.Accepted
          },
          registrationPathId: 'regPathId1',
          attendeeType: 'ATTENDEE',
          eventRegistrationId: '0bc2a08e-cde2-45f6-bc7f-d0f677013668'
        },
        'bcd51594-d0bb-4f51-9cdb-88f8e85addde': {
          attendee: {
            personalInformation: {
              firstName: 'Lonnie',
              lastName: 'Curtis'
            },
            inviteeStatus: InviteeStatusId.PendingApproval
          },
          registrationPathId: 'regPathId1',
          attendeeType: 'GROUP_LEADER',
          eventRegistrationId: 'bcd51594-d0bb-4f51-9cdb-88f8e85addde'
        },
        'c1f8b07a-ebd5-4051-be4a-15d6b5c96543': {
          attendee: {
            personalInformation: {
              firstName: 'Brittany',
              lastName: 'Richards'
            },
            inviteeStatus: InviteeStatusId.DeniedApproval
          },
          registrationPathId: 'regPathId1',
          attendeeType: 'ATTENDEE',
          eventRegistrationId: 'c1f8b07a-ebd5-4051-be4a-15d6b5c96543'
        },
        '82a42c25-d07b-4d84-afa5-73defb827c88': {
          attendee: {
            personalInformation: {
              firstName: 'Dwayne',
              lastName: 'Grant'
            }
          },
          registrationPathId: 'regPathId1',
          attendeeType: 'GUEST',
          eventRegistrationId: '82a42c25-d07b-4d84-afa5-73defb827c88'
        }
      }
    }
  }
};

describe('GroupRegistrationModificationDialog', () => {
  const store = createStoreWithMiddleware(
    combineReducers({
      dialogContainer,
      website: (x = {}) => x,
      text: (x = {}) => x,
      registrationForm: (x = {}) => x,
      appData: (x = {}) => x
    }),
    state
  );

  const dialog = mount(
    <Provider store={store}>
      <DialogContainer spinnerMessage="spinnerMessage" message="message" />
    </Provider>
  );

  test('matches snapshot when opened', () => {
    store.dispatch(openGroupRegistrationModificationDialog());
    expect(dialog).toMatchSnapshot();
  });

  test('eventRegistrations are present, except the ones denied approval and guests', () => {
    store.dispatch(openGroupRegistrationModificationDialog());
    dialog.update();
    expect(dialog.find('a[id="0bc2a08e-cde2-45f6-bc7f-d0f677013668"]').length).toBe(1);
    expect(dialog.find('a[id="bcd51594-d0bb-4f51-9cdb-88f8e85addde"]').length).toBe(1);
    expect(dialog.find('a[id="c1f8b07a-ebd5-4051-be4a-15d6b5c96543"]').length).toBe(0);
    expect(dialog.find('a[id="82a42c25-d07b-4d84-afa5-73defb827c88"]').length).toBe(0);
  });
});

describe('GroupRegistrationModificationDialog: with reg mod not allowed invitee', () => {
  const currState = {
    ...state,
    appData: {
      registrationSettings: {
        registrationPaths: {
          regPathId1: {
            modification: {
              enabled: false
            }
          }
        }
      }
    }
  };

  const store = createStoreWithMiddleware(
    combineReducers({
      dialogContainer,
      website: (x = {}) => x,
      text: (x = {}) => x,
      registrationForm: (x = {}) => x,
      appData: (x = {}) => x
    }),
    currState
  );

  const dialog = mount(
    <Provider store={store}>
      <DialogContainer spinnerMessage="spinnerMessage" message="message" />
    </Provider>
  );

  test('matches snapshot when opened', () => {
    store.dispatch(openGroupRegistrationModificationDialog());
    expect(dialog).toMatchSnapshot();
  });
});

describe('GroupRegistrationModificationDialog: with reg mod deadline passed invitee', () => {
  const currState = {
    ...state,
    appData: {
      registrationSettings: {
        registrationPaths: {
          regPathId1: {
            id: 'regPathId1',
            modification: {
              enabled: true,
              deadline: '2019-01-01T04:00:00.000Z'
            }
          }
        }
      }
    }
  };
  const store = createStoreWithMiddleware(
    combineReducers({
      dialogContainer,
      website: (x = {}) => x,
      text: (x = {}) => x,
      registrationForm: (x = {}) => x,
      appData: (x = {}) => x
    }),
    currState
  );
  const dialog = mount(
    <Provider store={store}>
      <DialogContainer spinnerMessage="spinnerMessage" message="message" />
    </Provider>
  );

  test('matches snapshot when opened', () => {
    store.dispatch(openGroupRegistrationModificationDialog());
    expect(dialog).toMatchSnapshot();
  });
});

describe('GroupRegistrationModificationDialog: pending approval modification disabled', () => {
  const currState = {
    ...state,
    appData: {
      registrationSettings: {
        registrationPaths: {
          regPathId1: {
            id: 'regPathId1',
            modification: {
              enabled: true,
              pendingEnabled: false
            }
          }
        }
      }
    }
  };
  const store = createStoreWithMiddleware(
    combineReducers({
      dialogContainer,
      website: (x = {}) => x,
      text: (x = {}) => x,
      registrationForm: (x = {}) => x,
      appData: (x = {}) => x
    }),
    currState
  );
  const dialog = mount(
    <Provider store={store}>
      <DialogContainer spinnerMessage="spinnerMessage" message="message" />
    </Provider>
  );

  test('matches snapshot when opened', () => {
    store.dispatch(openGroupRegistrationModificationDialog());
    expect(dialog).toMatchSnapshot();
  });

  test('eventRegistrations are present, only contain accepted invitee.', () => {
    store.dispatch(openGroupRegistrationModificationDialog());
    dialog.update();
    expect(dialog.find('a[id="0bc2a08e-cde2-45f6-bc7f-d0f677013668"]').length).toBe(1);
    expect(dialog.find('a[id="bcd51594-d0bb-4f51-9cdb-88f8e85addde"]').length).toBe(0);
    expect(dialog.find('a[id="c1f8b07a-ebd5-4051-be4a-15d6b5c96543"]').length).toBe(0);
    expect(dialog.find('a[id="82a42c25-d07b-4d84-afa5-73defb827c88"]').length).toBe(0);
  });
});
