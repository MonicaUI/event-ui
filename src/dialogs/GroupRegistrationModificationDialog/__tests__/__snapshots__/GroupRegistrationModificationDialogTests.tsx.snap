// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupRegistrationModificationDialog matches snapshot when opened 1`] = `
Array [
  <div
    class=""
  >
    <div
      class=""
    />
    <span
      class="mockNucleusCoreTransition"
    >
      <div
        class=""
      >
        <div
          aria-label="Dialog"
          role="dialog"
          style="outline: none;"
          tabindex="0"
        >
          <div
            class=""
            style="background-color: rgb(243, 59, 21); border-radius: 100px; position: absolute; left: 512px; top: 384px;"
          >
            <div
              class=""
            />
            <div
              class=""
            >
              <div
                class="dialogContainer"
              >
                <div
                  class="dialogChildren"
                >
                  <div
                    class="attendeeSelectionDialogContainer"
                  >
                    <div>
                      <div
                        aria-describedby="dialogHeader"
                        class=""
                        style="background-color: rgb(221, 226, 230); border-radius: 0;"
                      >
                        <div
                          class=""
                          id="dialogHeader"
                          style="background-color: rgb(249, 32, 242); padding: 15px 1.5625rem 15px 1.5625rem;"
                        >
                          <h2
                            class=""
                            style="color: rgb(255, 255, 255); font-family: Comic Sans MS, Comic Sans, Chalkboard, Marker Felt, sans-serif; font-size: 3.5625rem; font-style: normal; font-weight: 700; line-height: 1.3;"
                          >
                            <div
                              class=""
                              data-cvent-id="close"
                              role="button"
                              tabindex="0"
                            >
                              <span
                                class="container"
                              >
                                <span
                                  aria-hidden="true"
                                  class=""
                                  role="img"
                                />
                              </span>
                            </div>
                            EventGuestSide_GroupRegistrationModificationModal_SelectAttendee__resx
                          </h2>
                        </div>
                      </div>
                      <div
                        class=""
                        name="messageContainer"
                        style="background-color: rgb(243, 59, 21); padding: 0px 1.25rem 1.875rem 1.25rem;"
                        tabindex="-1"
                      >
                        <div>
                          <div
                            class=""
                          >
                            EventGuestSide_GroupRegistrationModificationModal_Instructions__resx
                          </div>
                          <div>
                            <div
                              class=""
                            >
                              <span
                                class="attendeeSelectionName"
                              >
                                <span
                                  class=""
                                  style="display: inline-block;"
                                >
                                  Lonnie Curtis
                                </span>
                              </span>
                              <span
                                class="attendeeSelectionLink"
                              >
                                <a
                                  class=""
                                  href="#"
                                  id="bcd51594-d0bb-4f51-9cdb-88f8e85addde"
                                  style="display: inline-block;"
                                >
                                  EventGuestSide_GroupRegistrationModificationModal_ModifyRegistration__resx
                                </a>
                              </span>
                            </div>
                            <div
                              class="attendeeSelectionSpace"
                            />
                          </div>
                          <div>
                            <div
                              class=""
                            >
                              <span
                                class="attendeeSelectionName"
                              >
                                <span
                                  class=""
                                  style="display: inline-block;"
                                >
                                  Ian Garcia
                                </span>
                              </span>
                              <span
                                class="attendeeSelectionLink"
                              >
                                <a
                                  class=""
                                  href="#"
                                  id="0bc2a08e-cde2-45f6-bc7f-d0f677013668"
                                  style="display: inline-block;"
                                >
                                  EventGuestSide_GroupRegistrationModificationModal_ModifyRegistration__resx
                                </a>
                              </span>
                            </div>
                            <div
                              class="attendeeSelectionSpace"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>,
]
`;

exports[`GroupRegistrationModificationDialog: pending approval modification disabled matches snapshot when opened 1`] = `
Array [
  <div
    class=""
  >
    <div
      class=""
    />
    <span
      class="mockNucleusCoreTransition"
    >
      <div
        class=""
      >
        <div
          aria-label="Dialog"
          role="dialog"
          style="outline: none;"
          tabindex="0"
        >
          <div
            class=""
            style="background-color: rgb(243, 59, 21); border-radius: 100px; position: absolute; left: 512px; top: 384px;"
          >
            <div
              class=""
            />
            <div
              class=""
            >
              <div
                class="dialogContainer"
              >
                <div
                  class="dialogChildren"
                >
                  <div
                    class="attendeeSelectionDialogContainer"
                  >
                    <div>
                      <div
                        aria-describedby="dialogHeader"
                        class=""
                        style="background-color: rgb(221, 226, 230); border-radius: 0;"
                      >
                        <div
                          class=""
                          id="dialogHeader"
                          style="background-color: rgb(249, 32, 242); padding: 15px 1.5625rem 15px 1.5625rem;"
                        >
                          <h2
                            class=""
                            style="color: rgb(255, 255, 255); font-family: Comic Sans MS, Comic Sans, Chalkboard, Marker Felt, sans-serif; font-size: 3.5625rem; font-style: normal; font-weight: 700; line-height: 1.3;"
                          >
                            <div
                              class=""
                              data-cvent-id="close"
                              role="button"
                              tabindex="0"
                            >
                              <span
                                class="container"
                              >
                                <span
                                  aria-hidden="true"
                                  class=""
                                  role="img"
                                />
                              </span>
                            </div>
                            EventGuestSide_GroupRegistrationModificationModal_SelectAttendee__resx
                          </h2>
                        </div>
                      </div>
                      <div
                        class=""
                        name="messageContainer"
                        style="background-color: rgb(243, 59, 21); padding: 0px 1.25rem 1.875rem 1.25rem;"
                        tabindex="-1"
                      >
                        <div>
                          <div
                            class=""
                          >
                            EventGuestSide_GroupRegistrationModificationModal_Instructions__resx
                          </div>
                          <div>
                            <div
                              class=""
                            >
                              <span
                                class="attendeeSelectionName"
                              >
                                <span
                                  class=""
                                  style="display: inline-block;"
                                >
                                  Ian Garcia
                                </span>
                              </span>
                              <span
                                class="attendeeSelectionLink"
                              >
                                <a
                                  class=""
                                  href="#"
                                  id="0bc2a08e-cde2-45f6-bc7f-d0f677013668"
                                  style="display: inline-block;"
                                >
                                  EventGuestSide_GroupRegistrationModificationModal_ModifyRegistration__resx
                                </a>
                              </span>
                            </div>
                            <div
                              class="attendeeSelectionSpace"
                            />
                          </div>
                          <div>
                            <div
                              class=""
                            >
                              EventGuestSide_GroupRegistrationModificationModal_ModificationNotAllowed__resx
                            </div>
                            <div>
                              <span
                                class="attendeeSelectionName"
                              >
                                <span
                                  class=""
                                  style="display: inline-block;"
                                >
                                  Lonnie Curtis
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>,
]
`;

exports[`GroupRegistrationModificationDialog: with reg mod deadline passed invitee matches snapshot when opened 1`] = `
Array [
  <div
    class=""
  >
    <div
      class=""
    />
    <span
      class="mockNucleusCoreTransition"
    >
      <div
        class=""
      >
        <div
          aria-label="Dialog"
          role="dialog"
          style="outline: none;"
          tabindex="0"
        >
          <div
            class=""
            style="background-color: rgb(243, 59, 21); border-radius: 100px; position: absolute; left: 512px; top: 384px;"
          >
            <div
              class=""
            />
            <div
              class=""
            >
              <div
                class="dialogContainer"
              >
                <div
                  class="dialogChildren"
                >
                  <div
                    class="attendeeSelectionDialogContainer"
                  >
                    <div>
                      <div
                        aria-describedby="dialogHeader"
                        class=""
                        style="background-color: rgb(221, 226, 230); border-radius: 0;"
                      >
                        <div
                          class=""
                          id="dialogHeader"
                          style="background-color: rgb(249, 32, 242); padding: 15px 1.5625rem 15px 1.5625rem;"
                        >
                          <h2
                            class=""
                            style="color: rgb(255, 255, 255); font-family: Comic Sans MS, Comic Sans, Chalkboard, Marker Felt, sans-serif; font-size: 3.5625rem; font-style: normal; font-weight: 700; line-height: 1.3;"
                          >
                            <div
                              class=""
                              data-cvent-id="close"
                              role="button"
                              tabindex="0"
                            >
                              <span
                                class="container"
                              >
                                <span
                                  aria-hidden="true"
                                  class=""
                                  role="img"
                                />
                              </span>
                            </div>
                            EventGuestSide_GroupRegistrationModificationModal_SelectAttendee__resx
                          </h2>
                        </div>
                      </div>
                      <div
                        class=""
                        name="messageContainer"
                        style="background-color: rgb(243, 59, 21); padding: 0px 1.25rem 1.875rem 1.25rem;"
                        tabindex="-1"
                      >
                        <div>
                          <div
                            class=""
                          >
                            EventGuestSide_GroupRegistrationModificationModal_Instructions__resx
                          </div>
                          <div>
                            <div
                              class=""
                            >
                              EventGuestSide_GroupRegistrationModificationModal_ModificationNotAllowed__resx
                            </div>
                            <div>
                              <span
                                class="attendeeSelectionName"
                              >
                                <span
                                  class=""
                                  style="display: inline-block;"
                                >
                                  Lonnie Curtis
                                </span>
                              </span>
                            </div>
                            <div>
                              <span
                                class="attendeeSelectionName"
                              >
                                <span
                                  class=""
                                  style="display: inline-block;"
                                >
                                  Ian Garcia
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>,
]
`;

exports[`GroupRegistrationModificationDialog: with reg mod not allowed invitee matches snapshot when opened 1`] = `
Array [
  <div
    class=""
  >
    <div
      class=""
    />
    <span
      class="mockNucleusCoreTransition"
    >
      <div
        class=""
      >
        <div
          aria-label="Dialog"
          role="dialog"
          style="outline: none;"
          tabindex="0"
        >
          <div
            class=""
            style="background-color: rgb(243, 59, 21); border-radius: 100px; position: absolute; left: 512px; top: 384px;"
          >
            <div
              class=""
            />
            <div
              class=""
            >
              <div
                class="dialogContainer"
              >
                <div
                  class="dialogChildren"
                >
                  <div
                    class="attendeeSelectionDialogContainer"
                  >
                    <div>
                      <div
                        aria-describedby="dialogHeader"
                        class=""
                        style="background-color: rgb(221, 226, 230); border-radius: 0;"
                      >
                        <div
                          class=""
                          id="dialogHeader"
                          style="background-color: rgb(249, 32, 242); padding: 15px 1.5625rem 15px 1.5625rem;"
                        >
                          <h2
                            class=""
                            style="color: rgb(255, 255, 255); font-family: Comic Sans MS, Comic Sans, Chalkboard, Marker Felt, sans-serif; font-size: 3.5625rem; font-style: normal; font-weight: 700; line-height: 1.3;"
                          >
                            <div
                              class=""
                              data-cvent-id="close"
                              role="button"
                              tabindex="0"
                            >
                              <span
                                class="container"
                              >
                                <span
                                  aria-hidden="true"
                                  class=""
                                  role="img"
                                />
                              </span>
                            </div>
                            EventGuestSide_GroupRegistrationModificationModal_SelectAttendee__resx
                          </h2>
                        </div>
                      </div>
                      <div
                        class=""
                        name="messageContainer"
                        style="background-color: rgb(243, 59, 21); padding: 0px 1.25rem 1.875rem 1.25rem;"
                        tabindex="-1"
                      >
                        <div>
                          <div
                            class=""
                          >
                            EventGuestSide_GroupRegistrationModificationModal_Instructions__resx
                          </div>
                          <div>
                            <div
                              class=""
                            >
                              EventGuestSide_GroupRegistrationModificationModal_ModificationNotAllowed__resx
                            </div>
                            <div>
                              <span
                                class="attendeeSelectionName"
                              >
                                <span
                                  class=""
                                  style="display: inline-block;"
                                >
                                  Lonnie Curtis
                                </span>
                              </span>
                            </div>
                            <div>
                              <span
                                class="attendeeSelectionName"
                              >
                                <span
                                  class=""
                                  style="display: inline-block;"
                                >
                                  Ian Garcia
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>,
]
`;
